airflow/models/pool.py

```diff
@@ -78,6 +86,54 @@ def get_default_pool(session: Session = None):
         """
         return Pool.get_pool(Pool.DEFAULT_POOL_NAME, session=session)
 
+    @staticmethod
+    @provide_session
+    def create_or_update_pool(name, slots, description, session=None):
```

Comment:
Let's use the new style now:

```suggestion
    def create_or_update_pool(name, slots, description, session=NEW_SESSION):
```